### PR TITLE
Resolved Implement OAM Services for version 2.0.0

### LINKED ISSUE #155: Implement OAM Services for version 2.0.0
### **Changes to the existing OAM services:**
**TcpClient :**
- [ ] Remove support for the getTcpClientRemoteIpv4Address , putTcpClientRemoteIpv4Address. Remove the functions from the controllers/TcpClient.js and service/TcpClientService.js
- [ ] Include support for the APIs getTcpClientRemoteAddress , putTcpClientRemoteAddress. Copy the functions from the generated server stub , paste it to the controllers/TcpClient.js and service/TcpClientService.js. Implement the service
- [ ] Include support for the APIs getTcpClientRemoteProtocol , putTcpClientRemoteProtocol . Copy the functions from the generated server stub , paste it to the controllers/TcpClient.js and service/TcpClientService.js. Implement the service

**HttpClient :**
- [ ] Include support for the API putHttpClientApplicationName . Copy the functions from the generated server stub , paste it to the controllers/HttpClient.js and service/HttpClientService.js. Implement the service

**TCPServer :**
- [ ] Remove support for the getTcpServerLocalIpv4Address, putTcpServerLocalIpv4Address. Remove the functions from the controllers/TcpServer.js and service/TcpServerService.js
- [ ] Include support for the APIs getTcpServerLocalAddress, putTcpServerLocalAddress. Copy the functions from the generated server stub , paste it to the controllers/TcpServer.js and service/TcpServerService.js. Implement the service
- [ ] Include support for the APIs getTcpServerLocalProtocol, putTcpServerLocalProtocol . Copy the functions from the generated server stub , paste it to the controllers/TcpServer.js and service/TcpServerService.js. Implement the service
- [ ] Include support for the APIs getTcpServerDescription, putTcpServerDescription. Copy the functions from the generated server stub , paste it to the controllers/TcpServer.js and service/TcpServerService.js. Implement the service

### **To include new functionalities of version 2.0.0 :**
**GenericResponseProfile :**
Support Version 2.0.0 changes for GenericResponseProfile
1. Generate server stub for the version 2.0.0 specification 
2. Copy the files controllers/GenericResponseProfile.js and service/GenericResponseProfileService.js from the generated stub to the source code.
3. If the api/openapi.yaml is not updated as per the published version 2.0.0 , then copy the api/openapi.yaml file from the generated stub to the source code
Following tasks needs to be completed to support GenericResponseProfile in version 2.0.0
Implement the logic in the controllers and services for the following services , 
- [ ] getGenericResponseProfileOperationName
- [ ] getGenericResponseProfileFieldName
- [ ] getGenericResponseProfileDescription
- [ ] getGenericResponseProfileDatatype
- [ ] getGenericResponseProfileValue
- [ ] putGenericResponseProfileValue

**ActionProfile :**
Support Version 2.0.0 changes for ActionProfile
1. Generate server stub for the version 2.0.0 specification 
2. Copy the files controllers/ActionProfileProfile.js and service/ActionProfileService.js from the generated stub to the source code.
3. If the api/openapi.yaml is not updated as per the published version 2.0.0 , then copy the api/openapi.yaml file from the generated stub to the source code
Following tasks needs to be completed to support ActionProfileProfile in version 2.0.0
Implement the logic in the controllers and services for the following services ,
- [ ] getActionProfileOperationName
- [ ] getActionProfileLabel
- [ ] getActionProfileInputValueListt
- [ ] getActionProfileDisplayInNewBrowserWindow
- [ ] getActionProfileConsequentOperationReference
- [ ] putActionProfileConsequentOperationReference

Core :
Include support for the API getProfileInstance. Copy the functions from the generated server stub , paste it to the controllers/TcpClient.js and service/TcpClientService.js. Implement the service.

### To include new Individual functionalities in version 2.0.0 :

IntegerProfile :
Support Version 2.0.0 changes for IntegerProfile
1. Generate server stub for the version 2.0.0 specification 
2. Copy the files controllers/IntegerProfile.js and service/IntegerProfileService.js from the generated stub to the source code.
3. If the api/openapi.yaml is not updated as per the published version 2.0.0 , then copy the api/openapi.yaml file from the generated stub to the source code

Following tasks needs to be completed to support IntegerProfile in version 2.0.0
Implement the logic in the controllers and services for the following services , 
- [ ] getIntegerProfileIntegerName
- [ ] getIntegerProfileUnit
- [ ] getIntegerProfileMinimum
- [ ] getIntegerProfileMaximum
- [ ] getIntegerProfileIntegerValue
- [ ] putIntegerProfileIntegerValue - Configuration range should be within the /integer-profile-capability/minimum and /integer-profile-capability/maximum

ElasticsearchClient :
- [ ] https://github.com/openBackhaul/OamLog/issues/107

### PR DESCRIPTION
**Pull request by ManasaBM1**
Fixes #155

### CODE REVIEW

**Comment by ManasaBM1** [comment]
Reviewer is @DanaSunal

File: server/controllers/ActionProfile.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
+'use strict';
+
+var utils = require('../utils/writer.js');
```
Comment (DanaSunal) [inline server/controllers/ActionProfile.js:3]: not used

**Review by DanaSunal — CHANGES_REQUESTED** [review]

File: server/service/TcpServerService.js [hunk 2]
```diff
@@ -80,14 +91,125 @@ exports.putTcpServerLocalIpv4Address = function(url, body) {
...
+  });
+}
+
+
+exports.getTcpServerDescription = function(url) {
+  return new Promise(async function(resolve, reject) {
+    var response = {};
+    var value = await fileOperation.readFromDatabaseAsync(url);
+    response['application/json'] = {
+  "tcp-server-interface-1-0:description" : "tcp-server-interface-1-0:"+value
```
[the remark this answers is not in the document]
Comment (ManasaBM1) [reply server/service/TcpServerService.js:207]: removed the "tcp-server-interface-1-0:

File: server/service/TcpServerService.js [hunk 3]
```diff
@@ -80,14 +91,125 @@ exports.putTcpServerLocalIpv4Address = function(url, body) {
...
+            forwardingAutomationInputList
+          );
+        }
+  
+        resolve();
+      } catch (error) {
+        reject();
+      }
+    });
+  }
```
[the remark this answers is not in the document]
Comment (ManasaBM1) [reply server/service/TcpServerService.js:173]: done

File: server/controllers/ActionProfile.js (hunk 1 above)
Comment (ManasaBM1) [reply server/controllers/ActionProfile.js]: removed done

**Review by DanaSunal — APPROVED** [review]

**Pull request merged**